phoenix-core/src/it/java/org/apache/phoenix/end2end/UngroupedAggregateRegionObserverSplitFailureIT.java

```diff
@@ -0,0 +1,107 @@
+import org.apache.hadoop.hbase.HBaseTestingUtility;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.coprocessor.ObserverContext;
+import org.apache.hadoop.hbase.regionserver.HRegion;
+import org.apache.phoenix.coprocessor.UngroupedAggregateRegionObserver;
+import org.apache.phoenix.jdbc.PhoenixDriver;
+import org.apache.phoenix.util.PhoenixRuntime;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class UngroupedAggregateRegionObserverSplitFailureIT extends BaseUniqueNamesOwnClusterIT {
+    private static HBaseTestingUtility hbaseTestUtil;
+    private static String zkQuorum;
+    private static String url;
+
+    @BeforeClass
+    public static synchronized void doSetup() throws Exception {
+        Configuration conf = HBaseConfiguration.create();
+        hbaseTestUtil = new HBaseTestingUtility(conf);
+        setUpConfigForMiniCluster(conf);
+        hbaseTestUtil.startMiniCluster();
+        // establish url and quorum. Need to use PhoenixDriver and not PhoenixTestDriver
+        zkQuorum = "localhost:" + hbaseTestUtil.getZkCluster().getClientPort();
+        url = PhoenixRuntime.JDBC_PROTOCOL + PhoenixRuntime.JDBC_PROTOCOL_SEPARATOR + zkQuorum;
+        DriverManager.registerDriver(PhoenixDriver.INSTANCE);
+    }
+
+    @AfterClass
+    public static synchronized void tearDownAfterClass() throws Exception {
+        if (hbaseTestUtil != null) {
+            hbaseTestUtil.shutdownMiniCluster();
+        }
+    }
+
+    @Test
+    public void testDeleteAfterSplitFailure() throws SQLException, IOException {
+        String tableName = generateUniqueName();
+        int numRecords = 100;
+        try (Connection conn = DriverManager.getConnection(url)) {
+            conn.createStatement().execute(
+                "CREATE TABLE " + tableName + " (PK1 INTEGER NOT NULL PRIMARY KEY, KV1 VARCHAR)");
+            int i = 0;
+            String upsert = "UPSERT INTO " + tableName + " VALUES (?, ?)";
+            PreparedStatement stmt = conn.prepareStatement(upsert);
+            while (i < numRecords) {
+                stmt.setInt(1, i);
+                stmt.setString(2, UUID.randomUUID().toString());
+                stmt.executeUpdate();
+                i++;
+            }
+            conn.commit();
+
+            List<HRegion> regions =
+                    hbaseTestUtil.getMiniHBaseCluster().getRegions(TableName.valueOf(tableName));
+            for (HRegion region : regions) {
+                UngroupedAggregateRegionObserver obs =
+                        (UngroupedAggregateRegionObserver) region.getCoprocessorHost()
+                                .findCoprocessor(UngroupedAggregateRegionObserver.class.getName());
+                ObserverContext ctx = new ObserverContext<>(null);
+                ctx.prepare(region.getCoprocessorHost().findCoprocessorEnvironment(
+                    UngroupedAggregateRegionObserver.class.getName()));
```

Comment:
nit: Since `preSplit` and `preRollBackSplit` requires `ObserverContext<RegionCoprocessorEnvironment>` specifically, we can define `ObserverContext` accordingly here:

```
                ObserverContext<RegionCoprocessorEnvironment> ctx = new ObserverContext<>(null);
                ctx.prepare((RegionCoprocessorEnvironment) region.getCoprocessorHost().findCoprocessorEnvironment(
                        UngroupedAggregateRegionObserver.class.getName()));
```